services/provider/server/server.go

```diff
@@ -779,3 +779,45 @@ func (s *OCSProviderServer) PeerBlockPool(ctx context.Context, req *pb.PeerBlock
 	}
 	return &pb.PeerBlockPoolResponse{}, nil
 }
+
+// RevokeBlockPoolPeering RPC call to delete the bootstrap secret to stop peering
+func (s *OCSProviderServer) RevokeBlockPoolPeering(ctx context.Context, req *pb.RevokeBlockPoolPeeringRequest) (*pb.RevokeBlockPoolPeeringResponse, error) {
+
+	klog.Infof("RevokeBlockPoolPeering request received for CephBlockPool %s and bootstrap secret %s", req.Pool, req.SecretName)
+
+	cephBlockPool, err := s.cephBlockPoolManager.GetBlockPoolByName(ctx, string(req.Pool))
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "Failed to find CephBlockPool resource %s: %v", req.Pool, err)
+	}
+
+	// delete secret and unset ref on the blockPool
+	err = s.cephBlockPoolManager.UnSetBootstrapSecretRef(ctx, req.SecretName, cephBlockPool)
+	// there might be a case where the bootstrap secret was deleted but request failed after this and there was a retry,
+	// if error is IsNotFound, that means it is safe to proceed as we have deleted the bootstrap secret
+	if err != nil && !kerrors.IsNotFound(err) {
+		return nil, status.Errorf(codes.Internal, "Failed to unset bootstrap secret ref for CephBlockPool resource %s: %v", req.Pool, err)
+	}
+
+	// disable mirroring on blockPool in the req
+	err = s.cephBlockPoolManager.DisableBlockPoolMirroring(ctx, cephBlockPool)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to disable mirroring for CephBlockPool resource %s: %v", req.Pool, err)
+	}
+
+	isRBDMirrorRequired, err := s.cephBlockPoolManager.IsRBDMirrorRequired(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to get if rbd mirror is required: %v,", err)
+	}
+
+	if !isRBDMirrorRequired {
```

Comment:
mirroring requires rbd-mirror and so you are deleting rbd-mirror if none of the cbp requires it, why not make the peering also consistent w/ this approach, like, client asks for peering but not explicitly enable mirroring as the latter is a hard requirement for the former?

Reply:
I am not sure I understand what you meant. For one-way mirroring to be set up, we need the following:
1. Side A (primary) should enable mirroring on their blockpool to have the bootstrap secret generated which will be used by Side B (secondary)
2. Side B enables the mirroring on their blockpool and set the bootstrap secret ref from side A
3. Side B deploys the rbd mirror